grails-app/services/org/pih/warehouse/report/NotificationService.groovy

```diff
@@ -245,10 +245,11 @@ class NotificationService {
     }
 
     void publishRequisitionStatusTransitionNotifications(Requisition requisition) {
-        List<Person> recipients = resolveNotificationRecipients(requisition)
-
         if (requisition.status == RequisitionStatus.PENDING_APPROVAL) {
+            List<Person> recipients = resolveNotificationRecipients(requisition)
             publishRequisitionPendingApprovalNotifications(requisition, recipients)
+        } else if (requisition.status in [RequisitionStatus.APPROVED, RequisitionStatus.REJECTED]) {
+            publishRequisitionStatusUpdateNotifications(requisition, requisition.requestedBy)
```

Comment:
I am not a big fan of using else, maybe this approach would be cleaner:
```groovy
void publishRequisitionStatusTransitionNotifications(Requisition requisition) {
        if (requisition.status == RequisitionStatus.PENDING_APPROVAL) {
            List<Person> recipients = resolveNotificationRecipients(requisition)
            publishRequisitionPendingApprovalNotifications(requisition, recipients)
            return
        }

        if (requisition.status in [RequisitionStatus.APPROVED, RequisitionStatus.REJECTED]) {
            publishRequisitionStatusUpdateNotifications(requisition, requisition.requestedBy)
        }
    }
```

Reply:
instead I went with an approach of switch/case